app_backend/llm_client.py

```diff
@@ -0,0 +1,345 @@
+        return {"model": model, "messages": messages, **opts}
+
+    def decode_response(self, response_data: Dict[str, Any]) -> str:
+        return response_data["choices"][0]["message"]["content"]
+
+    def extract_usage_info(self, response_data: Dict[str, Any]) -> Dict[str, Any]:
+        """Extract usage and cost information from OpenAI API response."""
+        usage_info = {}
+        
+        # Extract token usage
+        usage = response_data.get("usage", {})
+        if usage:
+            usage_info.update({
+                "prompt_tokens": usage.get("prompt_tokens", 0),
+                "completion_tokens": usage.get("completion_tokens", 0),
+                "total_tokens": usage.get("total_tokens", 0),
+            })
+        
+        # OpenAI now often includes cost directly in the response
+        if "cost" in response_data:
+            usage_info["cost_usd"] = response_data["cost"]
+        elif usage and "prompt_tokens_details" in usage:
+            # Some OpenAI responses include detailed cost breakdown
+            details = usage.get("prompt_tokens_details", {})
+            if "cost" in details:
+                usage_info["cost_usd"] = details["cost"]
+                
+        return usage_info
+
+
+class AnthropicProvider(LLMProvider):
+    """Anthropic API provider."""
+
+    def get_headers(self) -> Dict[str, str]:
+        return {"x-api-key": self.api_key, "anthropic-version": "2023-06-01"}
+
+    def encode_request(self, model: str, messages: List[Dict[str, str]], opts: Dict[str, Any]) -> Dict[str, Any]:
+        # Anthropic requires max_tokens
+        if "max_tokens" not in opts:
+            opts["max_tokens"] = 1024
+        return {"model": model, "messages": messages, **opts}
+
+    def decode_response(self, response_data: Dict[str, Any]) -> str:
+        return response_data["content"][0]["text"]
+
+    def extract_usage_info(self, response_data: Dict[str, Any]) -> Dict[str, Any]:
+        """Extract usage and cost information from Anthropic API response."""
+        usage_info = {}
+        
+        # Anthropic usage is in a different format
+        usage = response_data.get("usage", {})
+        if usage:
+            usage_info.update({
+                "prompt_tokens": usage.get("input_tokens", 0),
+                "completion_tokens": usage.get("output_tokens", 0),
+                "total_tokens": usage.get("input_tokens", 0) + usage.get("output_tokens", 0),
+            })
+        
+        # Anthropic may include cost in future API versions
+        if "cost" in response_data:
+            usage_info["cost_usd"] = response_data["cost"]
+            
+        return usage_info
+
+
+class LLMClient:
+    """
+    Client for LLM API operations with OpenAI, OpenRouter, and Anthropic support.
+    Follows the existing client pattern with dependency injection.
+    """
+
+    def __init__(
+        self,
+        http_client: httpx.Client,
+        openai_api_key: Optional[str] = None,
```

Comment:
One more parameter to optionally pass all the way from env var is probably a model name?